python_modules/dagster/dagster_tests/general_tests/check_tests/test_check.py

```diff
@@ -52,6 +52,14 @@ def test_bool_param():
         check.bool_param("val", "b")
 
 
+def test_bool_param_no_param_name():
+    my_value = 42
+    with pytest.raises(ParameterCheckError) as e:
+        check.bool_param(my_value)
+
+    assert str(e.value) == "Param \"my_value\" is not a bool. Got 42 which is type <class 'int'>."
```

Comment:
what's the pathological case here?  something like this?

```
def foo(a, b):
    check.opt_bool_param(a)
    check.bool_param(b)
    
foo(None, None)
```

Seems like this would be too easy to get wrong.